### PR TITLE
CC-5545: Side Nav

### PR DESCRIPTION
**Pull request by WenInCode**
### Description

Updates Consul UI to use [HDS::SideNav](https://helios.hashicorp.design/components/side-nav). Adds client-side search for datacenter/partition/namespace selectors.

Adds translations for side nav elements (not including login modal).
Uprgades HDS to latest release

Screenshot 2023-10-23 at 4 11 39 PM
Screenshot 2023-10-23 at 4 11 27 PM
Screenshot 2023-10-23 at 4 11 24 PM
Screenshot 2023-10-23 at 4 11 15 PM
Screenshot 2023-10-23 at 4 09 50 PM
Screenshot 2023-10-23 at 4 09 43 PM
Screenshot 2023-10-23 at 4 09 23 PM
Screenshot 2023-10-23 at 4 09 19 PM
Screenshot 2023-10-23 at 4 47 08 PM

### Links

https://hashicorp.atlassian.net/browse/CC-5545

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: ui/packages/consul-hcp/app/components/consul/hcp/home/index.hbs [hunk 1]
```diff
@@ -3,11 +3,14 @@
...
-<div
-  class="consul-hcp-home"
-  ...attributes
->
-  <a href={{env 'CONSUL_HCP_URL'}} data-native-href="true">
-    Back to HCP
-  </a>
-</div>
+{{#let @list (env "CONSUL_HCP_URL") as |SNL hcpUrl|}}
+  {{log hcpUrl}}
```
Comment (chris-hut) [inline ui/packages/consul-hcp/app/components/consul/hcp/home/index.hbs:7]:
```suggestion
```

File: ui/packages/consul-ui/app/locations/fsm-with-optional.js [hunk 2]
```diff
@@ -254,6 +254,10 @@ export default class FSMWithOptionalLocation {
    * performs an ember transition/refresh and browser location update using that
    */
   transitionTo(url) {
+    if (typeof this.router === 'undefined') {
+      this.router = this.container.lookup('router:main');
+    }
```
Comment (chris-hut) [inline ui/packages/consul-ui/app/locations/fsm-with-optional.js:259]: Mostly curious was this due to the change you made?
Or something that you just noticed along the way?

File: ui/packages/consul-ui/translations/components/consul/en-us.yaml [hunk 3]
```diff
@@ -70,6 +70,8 @@ acl:
         management: Management
         client: Client
 token:
+  selector:
+    logged-in:  "Logged in with ACL Token'<br><'b class=consul-side-nav__acl-logged-in'>'{token}'<'/b'>'"
```
Comment (chris-hut) [inline ui/packages/consul-ui/translations/components/consul/en-us.yaml:74]: Do we need all of the single quotes here?
I'm surprised it works tbh

**Review by chris-hut — APPROVED** [review]
QA looks great, should probably remove that log, but otherwise I love it!

File: ui/packages/consul-ui/app/locations/fsm-with-optional.js (hunk 2 above)
Comment (WenInCode) [reply ui/packages/consul-ui/app/locations/fsm-with-optional.js:259]: I just noticed that there was issues when trying to navigate to Docfy routes (eng docs). Sometimes the router would be null. Before we used Docfy generated link components but I wanted to just use the side nav stuff.

**Pull request merged**